pilot/pkg/config/clusterregistry/conversion.go

```diff
@@ -101,62 +100,140 @@ func (cs *ClusterStore) GetPilotClusters() (clusters []*k8s_cr.Cluster) {
 	return
 }
 
-// ReadClusters reads multiple clusters from files in a directory
-func ReadClusters(crPath string) (cs *ClusterStore, err error) {
+// ReadClusters reads multiple clusters from a OcnfigMap
+func ReadClusters(k8s kubernetes.Interface, configMapName string, configMapNamespace string, registriesDir string) (cs *ClusterStore, err error) {
 	cs = &ClusterStore{
 		clusters: []*k8s_cr.Cluster{},
 		cfgStore: nil,
 	}
-
 	var clusters []*k8s_cr.Cluster
-	err = filepath.Walk(crPath, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
+
+	// getClustersConfigs
+	clusters, err = getClustersConfigs(k8s, configMapName, configMapNamespace, registriesDir)
```

Comment:
TODO: watch

Reply:
@sbezverk the idea here is to use the watch API to dynamically update cluster configs - hence a TODO for the future.